### ISSUE TITLE
Support logIn with additional auth data

### ISSUE THREAD

**Issue by dblythy** [parse-community/Parse-SDK-JS#1807]
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/Parse-SDK-JS/issues?q=is%3Aissue).

### Current Limitation

Parse Server 6 supports logging in with a username/password, and additional required `authData`. This is not yet supported in this SDK

### Feature / Enhancement Description

Add login with additional authData support via `Parse.User.logIn("username", "password", authData)`

### Example Use Case

- 

### Alternatives / Workarounds

-

**Comment by chr4ss12** [comment]
is it currently not possible to login with username & password when you have an additional adapter?

```
logInWith('normal', {

authData: {
normal: {
  username: ...
  password:
},
challengeAdapter: {
   code: 2000
}
}
```

eg?

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.2.0-alpha.4](https://github.com/parse-community/Parse-SDK-JS/releases/tag/4.2.0-alpha.4)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.3.0-beta.1](https://github.com/parse-community/Parse-SDK-JS/releases/tag/4.3.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.3.0-alpha.1](https://github.com/parse-community/Parse-SDK-JS/releases/tag/4.3.0-alpha.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.3.0](https://github.com/parse-community/Parse-SDK-JS/releases/tag/4.3.0)
